src/synapse/api.py

```diff
@@ -147,5 +152,45 @@ def _get_nonce() -> str:
         requests.exceptions.Timeout,
         requests.exceptions.HTTPError,
     ) as exc:
-        logger.error("Failed to request %s : %s", REGISTER_URL, exc)
+        logger.exception("Failed to request %s : %r", REGISTER_URL, exc)
         raise NetworkError(f"Failed to request {REGISTER_URL}.") from exc
+
+
+def get_version() -> str:
+    """Get version.
+
+    Expected API output:
+    {
+        "server_version": "0.99.2rc1 (b=develop, abcdef123)",
+        "python_version": "3.7.8"
+    }
+
+    We're using retry here because after the config change, Synapse is restarted.
+
+    Returns:
+        The version returned by Synapse API.
+
+    Raises:
+        NetworkError: if there was an error fetching the version.
+    """
+    try:
+        session = Session()
+        retries = Retry(
+            total=3,
+            backoff_factor=3,
+        )
+        session.mount("http://", HTTPAdapter(max_retries=retries))
+        res = session.get(VERSION_URL, timeout=10)
+        res.raise_for_status()
+        server_version = res.json()["server_version"]
```

Comment:
Should the case where this lookup fails be handled gracefully? E.g., what if the `server_version` is not returned? Or no JSON is returned? Or a dict isn't returned?

Reply:
Sure, I added exceptions to this now.